src/screens/Account.jsx

```diff
@@ -3,20 +3,36 @@ import React, { useEffect, useState } from "react";
 import CandidateForm from "../components/CandidateForm";
 import PageTitle from "../components/PageTitle";
 import Loader from "../components/Loader";
-import { EDIT_CANDIDATE_FIELDS } from "../mocks";
+import { useContext } from "react";
+import { AppContext } from "../Context";
+import { editCandidate } from "../utils/request";
 
-const Account = () => {
+const Account = ({ updateUser }) => {
   // 1. Subscribe to AppContext data
-  const [fields, setFields] = useState(EDIT_CANDIDATE_FIELDS);
+  const userData = useContext(AppContext);
+  const [fields, setFields] = useState(null);
 
   useEffect(() => {
     // 2. Map with object keys and format your data before setting it in state
+    const user = userData.user.personal;
+    const formattedUser = Object.keys(user).map(key => ({
+      name: key,
+      value: user[key]
+    }));
+
+    setFields(formattedUser);
   }, []);
 
-  const onFormSubmit = (values) => {
-    console.log(values);
+
+  const onFormSubmit = async (values) => {
+    const id = userData.user.id;
+    const role = userData.user.role;
+
+    const res = await editCandidate(id, role, values);
```

Comment:
instead of passing role here and formatting your object in request.js. 

You could just do editCandidate({[userdata.role]:{...values}}), and then in reques.js you won't have to create a new reqBody object.